3-Authorization-II/protect-web-api/AppCreationScripts/README_STEPS.md

```diff
@@ -0,0 +1,471 @@
+public String date(BearerTokenAuthentication bearerTokenAuth) {
+    OAuth2AuthenticatedPrincipal principal = (OAuth2AuthenticatedPrincipal) bearerTokenAuth.getPrincipal();
+    return principal.getAttribute("scp").toString();
+}
+```
+
+### Protecting routes with AADWebSecurityConfigurerAdapter
+
+By default, this app protects all routes so that only users with a valid access token can access it. To configure your app's specific requirements, extend `AADWebSecurityConfigurationAdapter` in one of your classes. For an example, see this app's [SecurityConfig](.resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/SecurityConfig.java) class.
+
+This app also configures the correct claims validation for the incoming bearer token from the `app-id-uri` and `client-id` property the `application.yml` file.
+
+```java
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class SecurityConfig extends AADResourceServerWebSecurityConfigurerAdapter {
+    /**
+     * Add configuration logic as needed.
+     */
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        super.configure(http);
+        http.authorizeRequests((requests) -> requests.anyRequest().authenticated());
+    }
+}
+```
+
+### Validate your Azure access tokens using routes with AADDelegatingOAuth2TokenValidator
+
+While `AADWebSecurityConfigurationAdapter` can be used to protect your routes so only valid users can access it, Azure protected API's require additional claims validation to ensure only valid users are calling you routes. For this, see this app's [AppSecurityConfig](.resource-api/src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapi/AppSecurityConfig.java) class.
+
+This class configures your `AADWebSecurityConfigurationAdapter` with signature validation using `NimbusJwtDecoder` as well as custom validation for the `iss`, `aud`, `nbf`, and `exp` claims using `AADDelegatingOAuth2TokenValidator`
+
+```java
+    @Bean
+    JwtDecoder jwtDecoder() {
+    	String JWKSet = instanceUri + tenantId + "/discovery/v2.0/keys";
+        NimbusJwtDecoder nimbusJwtDecoder = NimbusJwtDecoder.withJwkSetUri(JWKSet).build();
+        nimbusJwtDecoder.setJwtValidator(jwtValidator());
+        return nimbusJwtDecoder;
+    }
+```
+Additionally, you may also configure this class to perform custom extended claim validation of the `azp` and `appid` claim to restrict access of your API to only select web applications. 
+```java
+	/*Extended Validation 
+	 * Uncomment to allow user to limit access of API to specific client apps
+	 * Add client Id of your client apps to allowedClientApps
+	 */
+	String[] allowedClientApps = new String[] {""};
+	for (int i = 0; i < allowedClientApps.length; i++ ) {
+		validators = AADHelpers.AADExtendedValidators(validators, allowedClientApps[i]);
+	}
+```	
+
+## How to deploy this sample to Azure
+
+<details>
```

Comment:
@kalyankrishna1 I think in codegen we need a sentence here that says something like "there are two projects here, each deployed seperately..." for clarity